### ISSUE TITLE
Publish to chromatic GitHub action failing

### ISSUE THREAD

**Issue by ccrowley96** [chromaui/chromatic-cli#349]
I've attached the GitHub Action log at the bottom of the issue.  

This problem seems to be with Node Sass.  

Here is the `chromatic.yml` GitHub Action.

```
# Workflow name
name: 'Chromatic'

# Event for the workflow
on: push

jobs:
    chromatic-deployment:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v2
              with:
                  fetch-depth: 0 # 👈 Required to retrieve git history
            - uses: actions/setup-node@v1
            - name: Cache node modules
              uses: actions/cache@v2
              env:
                  cache-name: cache-node-modules
              with:
                  # npm cache files are stored in `~/.npm` on Linux/macOS
                  path: ~/.npm
                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
                  restore-keys: |
                      ${{ runner.os }}-build-${{ env.cache-name }}-
                      ${{ runner.os }}-build-
                      ${{ runner.os }}-

            - name: Install dependencies
              run: npm install

            - name: Publish to chromatic
              uses: chromaui/action@v1
              # Chromatic GitHub Action options
              with:
                  token: ${{ secrets.GITHUB_TOKEN }}
                  # 👇 Chromatic projectToken, refer to the manage page to obtain it.
                  projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
                  exitZeroOnChanges: true # 👈 Option to prevent the workflow from failing
                  exitOnceUploaded: true # 👈 Exit with status 0 (OK) once the build has been sent to Chromatic
                  buildScriptName: storybook:export

```

I've tried adding `npm rebuild node-sass` and changing the node version to explicitly use Node version 10, but I'm still getting the same error.

```
- uses: actions/setup-node@v2
   with:
     node-version: '10'
```

### Error snippet

```
ERR! => Failed to build the preview
ERR! ./src/Components/BIMViewer/BIMViewer.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js??ref--14-3!./src/Components/BIMViewer/BIMViewer.scss)
ERR! Module Error (from ./node_modules/sass-loader/dist/cjs.js):
ERR! Missing binding /home/runner/work/iot-cardboard-js/iot-cardboard-js/node_modules/node-sass/vendor/linux-x64-72/binding.node
ERR! Node Sass could not find a binding for your current environment: Linux 64-bit with Node.js 12.x
ERR! 
ERR! Found bindings for the following environments:
ERR!   - Linux 64-bit with Node.js 10.x
ERR! 
ERR! This usually happens because your environment has changed since running `npm install`.
ERR! Run `npm rebuild node-sass` to download the binding for your current environment.
```

### Chromatic action step (up to first failure)

```
Run chromaui/action@v1
  with:
    token: ***
    projectToken: ***
    exitZeroOnChanges: true
    exitOnceUploaded: true
    buildScriptName: storybook:export

Chromatic CLI v5.8.2
https://www.chromatic.com/docs/cli

Authenticating with Chromatic
    → Connecting to https://index.chromatic.com
Authenticated with Chromatic
    → Using project token '********8661'
Retrieving git information
Retrieved git information
    → Commit 'b715b6a' on branch 'user/niusoff/xeokit'; found 1 parent commit
Collecting Storybook metadata
Collected Storybook metadata
    → Storybook v6.1.21 for React; supported addons found: Essentials
Building your Storybook
    → Running command: npm run --silent storybook:export -- --output-dir /tmp/chromatic--1766-v7HIDElUi8Ar
The CLI tried to run your storybook:export script, but the command failed. This indicates a problem with your Storybook. Here's what to do:

- Check the Storybook build log printed below.
- Run npm run storybook:export or yarn storybook:export yourself and make sure it outputs a valid Storybook by opening the generated index.html in your browser.
- Review the build-storybook CLI options at https://storybook.js.org/docs/configurations/cli-options/#for-build-storybook

Command failed with exit code 1: npm run --silent storybook:export -- --output-dir /tmp/chromatic--1766-v7HIDElUi8Ar

ℹ Storybook build output:
/home/runner/work/iot-cardboard-js/iot-cardboard-js/build-storybook.log

info @storybook/react v6.1.21
info 
info => Cleaning outputDir /tmp/chromatic--1766-v7HIDElUi8Ar
info => Building manager..
info => Loading manager config..
info => Loading presets
info => Compiling manager..
info => Manager built (21 s)
info => Building preview..
info => Loading preview config..
info => Loading presets
info => Loading 1 config file in "./.storybook"
info => Loading 6 other files in "./.storybook"
info => Adding stories defined in ".storybook/main.js"
info => Using default Webpack setup
info => Compiling preview..
ERR! => Failed to build the preview
ERR! ./src/Components/BIMViewer/BIMViewer.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js??ref--14-3!./src/Components/BIMViewer/BIMViewer.scss)
ERR! Module Error (from ./node_modules/sass-loader/dist/cjs.js):
ERR! Missing binding /home/runner/work/iot-cardboard-js/iot-cardboard-js/node_modules/node-sass/vendor/linux-x64-72/binding.node
ERR! Node Sass could not find a binding for your current environment: Linux 64-bit with Node.js 12.x
ERR! 
ERR! Found bindings for the following environments:
ERR!   - Linux 64-bit with Node.js 10.x
ERR! 
ERR! This usually happens because your environment has changed since running `npm install`.
ERR! Run `npm rebuild node-sass` to download the binding for your current environment.

...
...
...
...

Error: non-zero exit code
```

**Comment by shadiramadan** [comment]
https://github.com/chromaui/action doesn't allow posting issues.
Our chromatic builds started failing > 4 hours ago.

We get a slightly different error- but it looks like the https://github.com/chromaui/action repo had a force push on the master branch- there is only one commit that is 6 hours old.

**Comment by shadiramadan** [comment]
@ghengeveld

**Comment by tmeasday** [comment]
@shadiramadan this is the correct repo to be posting issues w/ the action. 

> We get a slightly different error

What error do you get?

**Comment by imgos** [comment]
I'm also getting a related error, which started at the same time.  It seems to have to do with the action somehow using a different version of node (12.13.1) than expected.  The version of node in `/usr/local/bin/node` is 14.17.0).  I've also tried `actions/setup-node`, which is currently giving me 14.16.1 and also results in the same error on the chromaui action.

Our `package.json` pins the node engine to "14", so we get this error when running the chromaui action:

```
ℹ Storybook build output:
/home/runner/work/my-repo-name/build-storybook.log

yarn run v1.22.10
error my-package-name@0.1.0: The engine "node" is incompatible with this module. Expected version "14". Got "12.13.1"
error Commands cannot run with an incompatible environment.
```

**Comment by ccrowley96** [comment]
This is our latest `chromatic.yml` action - seems to have fixed the issue for us (for the time being).

We switched to `actions/setup-node@v2` and `node-version: '12.x'`

We also removed the npm file caching.

```yml
# Workflow name
name: 'Chromatic'

# Event for the workflow
on: push

jobs:
    chromatic-deployment:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v2
              with:
                  fetch-depth: 0 # 👈 Required to retrieve git history
            - uses: actions/setup-node@v2
              with:
                  node-version: '12.x'

            - name: Install dependencies
              run: npm install

            - name: Publish to chromatic
              uses: chromaui/action@v1
              # Chromatic GitHub Action options
              with:
                  token: ${{ secrets.GITHUB_TOKEN }}
                  # 👇 Chromatic projectToken, refer to the manage page to obtain it.
                  projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
                  exitZeroOnChanges: true # 👈 Option to prevent the workflow from failing
                  exitOnceUploaded: true # 👈 Exit with status 0 (OK) once the build has been sent to Chromatic
                  buildScriptName: storybook:export
```

**Comment by shadiramadan** [comment]
@tmeasday I get:
```
Run chromaui/action@v1

Chromatic CLI v5.8.2
https://www.chromatic.com/docs/cli

Authenticating with Chromatic
    → Connecting to https://index.chromatic.com
Authenticated with Chromatic
    → Using project token '********eb28'
Retrieving git information
Retrieved git information
    → Commit '1225b5b' on branch 'feat/authModalBuild'; found 1 parent commit
Collecting Storybook metadata
Collected Storybook metadata
    → Storybook v6.1.17 for React; supported addons found: Essentials, Info, Links
Building your Storybook
    → Running command: npm run --silent build-storybook -- --output-dir /tmp/chromatic--952-WoKwSey3VPGM
The CLI tried to run your build-storybook script, but the command failed. This indicates a problem with your Storybook. Here's what to do:

- Check the Storybook build log printed below.
- Run npm run build-storybook or yarn build-storybook yourself and make sure it outputs a valid Storybook by opening the generated index.html in your browser.
- Review the build-storybook CLI options at https://storybook.js.org/docs/configurations/cli-options/#for-build-storybook

Command failed with exit code 1: npm run --silent build-storybook -- --output-dir /tmp/chromatic--952-WoKwSey3VPGM

ℹ Storybook build output:
/runner/_work/react-component-library/react-component-library/build-storybook.log

internal/modules/cjs/loader.js:800
    throw err;
    ^
Error: Cannot find module '../lib/utils/unsupported.js'
Require stack:
- /runner/externals.2.278.0/node12/bin/npm
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
    at Function.Module._load (internal/modules/cjs/loader.js:690:27)
```

This is my workflow yaml
```yaml
name: Build / Publish to Chromatic

on:
  push:
    branches:
      - '**'

jobs:
  build-publish:
    runs-on: [self-hosted, Linux, X64]

    steps:
    - name: Checkout
      uses: actions/checkout@v2
      # Required to retrieve git history for Chromatic
      # https://www.chromatic.com/docs/github-actions#support-for-codeactionscheckoutv2code
      with:
        fetch-depth: 0

    - name: Use Node.js
      uses: actions/setup-node@v2
      with:
        node-version: '14'

    - name: Install Dependencies
      run: |-
        npm install -g yarn
        yarn

    # - name: Lint
    #   run: yarn lint

    - name: Test
      run: yarn test --ci --coverage --maxWorkers=2

    - name: Build
      run: yarn build

    - name: Publish to Chromatic
      uses: chromaui/action@v1
      with:
        token: ${{ secrets.GITHUB_TOKEN }}
        projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
```

The logs for the `Use Node.js` step
```
Run actions/setup-node@v2
Attempting to download 14...
Acquiring 14.17.0 - x64 from https://github.com/actions/node-versions/releases/download/14.17.0-834391622/node-14.17.0-linux-x64.tar.gz
Extracting ...
/usr/bin/tar xz --strip 1 --warning=no-unknown-keyword -C /runner/_work/_temp/7b910c46-5e61-4520-bed5-fea828e6acfe -f /runner/_work/_temp/3aab1fe8-2fcd-4fdc-aefb-0a8062a858bf
Adding to the cache ...
Done
```

**Comment by tmeasday** [comment]
Can folks try `chromaui/action-canary@v1`? It appears to be fixing the remaining issues for some users, would be good to confirm before upgrading to the next version.

**Comment by ghengeveld** [comment]
We've just released v5.8.3 and updated `chromaui/action@v1` accordingly. You should switch back to the regular action if you were using the canary.
